### ISSUE TITLE
Enhancement for Plug-in support: print out warning message when plug-ins overwrite the same object

### ISSUE THREAD

**Issue by pega3** [endless-sky/endless-sky#5189]
**Is your feature request related to a problem? Please describe.**
Currently, plug-ins can overwrite the same object without realizing this, potentially breaking one of the two involved plug-ins. While this so far has been a rare case (Akasha / Enclave), this feature makes it easier for plug-in developers to avoid collisions. 
In theory, that could also happen within the game data, however, it can be expected that this is detected by the developers very quickly.

**Describe the solution you'd like**
The loading of game data is happening in stages. If within one stage the same object is defined twice, a warning should be written to the log. Ideally, the warning would quote all locations of the redundant definition.
In a nutshell: if the order in which the files of a given stage are loaded affects the result, something is not good. 

**Additional context**
The case quoted above were a result of both plug-in authors using the same world generator (Nucleartaxi / endless-sky-generators), which has a limited name space for the object it creates.
In general, this is certainly not a pressing issue, but a "nice to have".

**Comment by tehhowch** [comment]
This is something that would be a part of rich plugin support via an in-game plugin manager. In other related issues, I've identified that such a manager would be able to identify which objects a plugin exports, and would display that information to the user. Thus this situation would be explicitly apparent to the user, e.g. the enabling plugin x will introduce new object A and modify existing object B.

It is expected to be a normal use case that plugins will modify core game data, and it is similarly allowed for a plugin to fix or extend some other plugin, so I don't feel it is appropriate to display a warning when this occurs.

**Comment by pega3** [comment]
@tehhowch - thank you for your comment. How can a plug-in fix or extend another plug-in in the current game? Is there some intelligent parsing happening, which defines the load order, or is it "by default the last one wins"?

I agree that in the scenario you describe as rich plugin support, the information would be displayed to the user, but in case of a larger plug-in which introduces dozens of systems, planets, ships etc, that information might easily be buried. Then again, while average user likely won't look at that display, the plug-in developer hopefully will.

Is what you describe as rich plugin support documented somewhere?
In that case, I'll happily close this issue.
